### PR TITLE
Organizer registration

### PR DESCRIPTION
**Pull request by jaBrzosko**
Endpoint for organizer to register with validation and tests

### CODE REVIEW

File: src/back/TicketManager/src/Core/TicketManager.Core.Contracts/Organizers/RegisterOrganizerRequest.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        public const int TaxIdIsEmpty = 13;
+        public const int TaxIdIsTooLong = 14;
+        public const int DisplayNameIsEmpty = 15;
+        public const int DisplayNameIsTooLong = 16;
+        public const int PhoneNumberIsEmpty = 17;
+        public const int PhoneNumberIsTooLong = 18;
+    }
+}
+
+public enum TaxIdEnum
```
Comment (michal12334) [inline src/back/TicketManager/src/Core/TicketManager.Core.Contracts/Organizers/RegisterOrganizerRequest.cs:36]:
```suggestion
public enum TaxIdTypeDto
```

File: src/back/TicketManager/src/Core/TicketManager.Core.Domain/TicketManager.Core.Domain.csproj [hunk 2]
```diff
@@ -7,4 +7,8 @@
         <WarningsAsErrors>Nullable</WarningsAsErrors>
     </PropertyGroup>
 
+    <ItemGroup>
+      <ProjectReference Include="..\TicketManager.Core.Contracts\TicketManager.Core.Contracts.csproj" />
+    </ItemGroup>
```
Comment (michal12334) [inline src/back/TicketManager/src/Core/TicketManager.Core.Domain/TicketManager.Core.Domain.csproj:12]: The reason why `domain` and `contracts` are separate projects is to not allow that. Generally classes in `domain` and in `contracts` have different responsibilities and shouldn't be the same classes. Sometimes result of that is that we have to write same classes/enums twice.

File: src/back/TicketManager/src/Core/TicketManager.Core.Services/DataAccess/CoreDbContext.cs [hunk 3]
```diff
@@ -34,6 +38,22 @@ private void ConfigureUsers(ModelBuilder modelBuilder)
...
     
+    private void ConfigureOrganizers(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Organizer>(cfg =>
+        {
+            cfg.HasKey(e => e.Id);
+            cfg.Property(e => e.Email).HasMaxLength(StringLengths.MediumString);
+            cfg.Property(e => e.CompanyName).HasMaxLength(StringLengths.ShortString);
+            cfg.Property(e => e.TaxId).HasMaxLength(StringLengths.ShortString);
+            cfg.Property(e => e.Address).HasMaxLength(StringLengths.MediumString);
```
Comment (michal12334) [inline src/back/TicketManager/src/Core/TicketManager.Core.Services/DataAccess/CoreDbContext.cs:49]: I'm not sure if address can be that long, but keep whatever you want.

File: src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/RegisterOrganizerValidator.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+using TicketManager.Core.Services.Services.Mockables;
+using TicketManager.Core.Services.ValidationExtensions;
+
+namespace TicketManager.Core.Services.Endpoints.Organizers;
+
+public class RegisterOrganizerValidator: Validator<RegisterOrganizerRequest>
+{
+    private readonly IServiceScopeFactory scopeFactory;
+    private readonly MockableCoreDbResolver dbResolver;
+    public RegisterOrganizerValidator(IServiceScopeFactory scopeFactory, MockableCoreDbResolver dbResolver)
```
Comment (michal12334) [inline src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/RegisterOrganizerValidator.cs:17]: Add new line here.

File: src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/RegisterOrganizerValidator.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        
+        RuleFor(req => req.PhoneNumber)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.PhoneNumberIsEmpty)
+            .MaximumLength(StringLengths.ShortString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.PhoneNumberIsTooLong);
+        
+    }
+    
+    private Task<bool> IsEmailAvailable(string email, CancellationToken cancellationToken)
```
Comment (michal12334) [inline src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/RegisterOrganizerValidator.cs:72]: Add `async` sufix here. I guess it should be also added in `RegisterUserValidator`. Also I'm thinking if we should extract this method somewhere else to make it available for both validators and write it only once. But until this method is small and written only twice both solutions are ok.

File: src/back/TicketManager/tests/Core/TicketManager.Core.ServicesTests/Endpoints/Organizers/RegisterOrganizerValidatorTests.cs [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+            CompanyName = "company sp. z o. o.",
+            Address = "ul. Koszykowa 75, 00-662 Warszawa, Poland",
+            TaxId = "3423272256",
+            TaxIdType = TaxIdEnum.Nip,
+            DisplayName = "Super company",
+            Email = "email@emai.com",
+            Password = "Password1",
+            PhoneNumber = "123-456-789"
+        };
+        singleEntryValidator = GetValidator(new List<Account>() { new(Guid.NewGuid(), "email2@email.com", "passwordHash", AccountRoles.Organizer) });
```
Comment (michal12334) [inline src/back/TicketManager/tests/Core/TicketManager.Core.ServicesTests/Endpoints/Organizers/RegisterOrganizerValidatorTests.cs:34]: That's good idea to create it here for all tests.

**Review by michal12334 — CHANGES_REQUESTED** [review]
The biggest issue here is dependency between `contracts` and `domain`. Also configure your ide to add new lines at the end of files. I'll try to configure it for the project for all programmers and we'll see if it's gonna work.

File: src/back/TicketManager/src/Core/TicketManager.Core.Domain/TicketManager.Core.Domain.csproj (hunk 2 above)
Comment (jaBrzosko) [reply src/back/TicketManager/src/Core/TicketManager.Core.Domain/TicketManager.Core.Domain.csproj:12]: Yes, sure. TaxIdType enum got mixed up between those two projects

**Pull request merged**